### ISSUE TITLE
Shadowing macro definitions should warn at least

### ISSUE THREAD

**Issue by lifthrasiir** [rust-lang/rust#20192]
```rust
#![feature(macro_rules)]
macro_rules! a((foo) => ("foo"));
macro_rules! a((bar) => ("bar")); // does not warn
```

Inspired by #20191. Not sure how to disable the warning when required (e.g. for rustdoc) though.

**Comment by steveklabnik** [comment]
Since new lints have a big impact on users of rustc, the policy is that they should go through the RFC process like other user-facing changes. As such, I'm going to give this one a close, but if anyone comes across this ticket and wants this lint, consider adding it to clippy and/or writing up an RFC. Thanks!
